envoy: log version mismatch only after failing twice

marseel opened the pull request:
Previously, we were logging version mismatch as an error each time that probe failed.
However, this does not make sense, as during regular minor or patch version upgrades, it's possible that Envoy daemonset version and Agent version do not match for brief period of time during upgrade. This can happen either when Envoy DS is upgraded first or when Agent is upgraded first on a particular node.

Now, if we log error about version mismatch, we know that mismatch has been happening for at least 5 minutes.


```release-note
envoy: Cilium does no longer report brief version mismatch of Agent and Envoy during upgrade
```

marseel committed:
envoy: log version mismatch only after failing twice

Previously, we were logging version mismatch as an error each time that
probe failed.
However, this does not make sense, as during regular minor or patch
version upgrades, it's possible that Envoy daemonset version and
Agent version do not match for brief period of time during upgrade.
This can happen either when Envoy DS is upgraded first or when Agent is
upgraded first on a particular node.
Additionally, Increase probing frequecy from 5 minutes to 2 minutes.

Now, if we log error about version mismatch, we know that mismatch has
been happening for at least 2 minutes.

Signed-off-by: Marcel Zieba <marcel.zieba@isovalent.com>

sayboras commented on .github/workflows/tests-e2e-upgrade.yaml:
```diff
@@ -379,7 +379,6 @@ jobs:
         shell: bash
         run: |
           cilium upgrade \
-            --helm-set=disableEnvoyVersionCheck=true \
```
if we remove this one, might need to allow error log for version mismatch in cilium cli check error log.

marseel replied:
Why? I don't think we are expecting version mismatch to happen for prolonged time, no?

sayboras replied:
it's more about our test setup, cilium upgrade command might not use the upgraded helm chart (with new cilium-envoy version), the the mismatch will happen.

https://github.com/cilium/cilium/commit/4a461aeb88a3de42d4b43b3b2a8c9dbfa5cf7217

marseel replied:
Apologies, but I still don't get it :cry: 

What I can see is:
```
      - name: Upgrade Cilium
        if: ${{ matrix.skip-upgrade != 'true' }}
        shell: bash
        run: |
          cilium upgrade \
          ${{ steps.cilium-newest-config.outputs.config }} \
```
which should point to new helm chart:
```
      - name: Derive newest Cilium installation config
     [...]
          chart-dir: './untrusted/cilium-newest/install/kubernetes/cilium'
```
with upgraded Envoy?

marseel replied:
Okay, I dropped disabling version check for now for tests.

marseel committed:
envoy: do not retry probes

As we log error only after second failed probe, we no longer need to
retry multiple times when the probe fails.
A single probe failure will be logged as info log, which can also give
additional insights into health of Envoy.

Signed-off-by: Marcel Zieba <marcel.zieba@isovalent.com>

mhofstetter reviewed (approved):
Thanks!
